### ISSUE TITLE
Post Training quantization knowledge

### ISSUE THREAD

**Issue by joaolcaas** [NervanaSystems/distiller#159]
Hi!
First of all, i'm running distiller on a GTX 750 Ti, i7 and 8gb ram.
  
I had made some tests on distiller (more specif on quantization) and i'm having some troubles to understand the results that you guys can see in #137, more specif about the test time on quantized model:

- Time on pre-trained: 0.405s
- Time on quantized: 6.814s

As we can see, the test time on pre-trained model is much slower than quantized, while distiller is proposed to [increase inference speed](https://nervanasystems.github.io/distiller/).
In one research that my team had made, we found something very interesting that Neta wrote about one of his researches on distiller:

![image](https://user-images.githubusercontent.com/23393117/52970999-90430b00-3394-11e9-81a5-8b5bf253137f.png)

We understood that post training is on step 2, considering  redundant operations and thats cause the large time difference between the two model above.
That image can be applied to post training? And if the answer is positive, distiller already suport it? Or that is for coming next?

Thanks!

**Comment by guyjacob** [comment]
Hi @joaolcaas,

As it is today, Distiller only **simulates** post-training quantization. This allows us to study the effect of quantization on accuracy. It does not, unfortunately, provide insights on the actual runtime of the quantized model.

We did not implement low-level specialized operations that utilize 8-bit capabilities of the CPU/GPU.
Specifically, the way post-training quantization is implemented, supported layers are wrapped with quantize/de-quantizer operations, but the layers themselves are unchanged. They still operate on FP32 tensors, but these tensors are restricted to contain only integer values.
So, as you can see, we are only adding operations in order to simulate quantization. Therefore it isn't surprising at all that you're getting slower runtime when quantizing.

We do not have plans to implement "native" low-precision operations within Distiller. We do, however, plan to support exporting of quantized models using ONNX, once quantization is published as part of the standard. Then you'll be able to export a model quantized in Distiller and run it on a framework that supports actual 8-bit execution on CPU/GPU. In addition, 8-bit support in PyTorch itself is also in the works (it's already implemented in Caffe2). Once that's released, we'll see how we can integrate Distiller with that.

I'm sorry for the mis-understanding, and I hope this clears everything.

**Comment by joaolcaas** [comment]
I see. Thank you!

**Comment by nzmora** [comment]
Closing this issue as it's been use for a long time.
